pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go

```diff
@@ -0,0 +1,333 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package assemble
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha2"
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+	ctrlutil "github.com/oam-dev/kubevela/pkg/controller/utils"
+	"github.com/oam-dev/kubevela/pkg/oam"
+	"github.com/oam-dev/kubevela/pkg/oam/util"
+)
+
+// NewAssembleOptions create a Options
+func NewAssembleOptions(appRevision *v1beta1.ApplicationRevision) *Options {
+	return &Options{AppRevision: appRevision}
+}
+
+// Options contains configuration to assemble resources recorded in the ApplicationRevision.
+// 'Assemble' means completing workload/trait resources and get them ready to emit into K8s
+// cluster.
+type Options struct {
+	AppRevision     *v1beta1.ApplicationRevision
+	WorkloadOptions []WorkloadOption
+
+	AppConfig      *v1alpha2.ApplicationConfiguration
+	Comps          []*v1alpha2.Component
```

Comment:
```suggestion
	V1alpha2AppConfig      *v1alpha2.ApplicationConfiguration
	V1alpha2Comps          []*v1alpha2.Component
```

Does it necessary？ I think it would be deprecated？ Let's make it clear if Necessary

Reply:
I made a helper struct to replace AppConfig just in assemble pkg.  As for v1alpha2.Component and ComponentTrait, it costs a lot to replace them now. Maybe we can refactor them later.